src/vs/workbench/contrib/terminal/browser/terminalService.ts

```diff
@@ -1174,10 +1178,15 @@ export class TerminalService extends Disposable implements ITerminalService {
 		this._terminalGroupService.createGroup(instance);
 
 		// Make active automatically if it's the first instance
-		if (this.instances.length === 1) {
+		if (this.instances.length === 1 && !suppressSetActive) {
 			this._terminalGroupService.setActiveInstanceByIndex(0);
 		}
 
+		if (forceSaveState) {
+			// Skips the debounce of _saveState as we are shutting down
+			// and need to save the revealed hideFromUser terminals
+			await this._saveStateNow();
+		}
```

Comment:
`showBackgroundTerminal` doesn't imply it's shutting down. It's fine to do this, just should tweak to comment to "in case it's shutting down" or similar.